src/backend/commands/tablecmds.c

```diff
@@ -18774,6 +18785,12 @@ make_distributedby_for_rel(Relation rel)
 
 	dist = makeNode(DistributedBy);
 
+	if (Gp_role == GP_ROLE_UTILITY)
+	{
+		Assert(policy->ptype == POLICYTYPE_ENTRY);
+		return NULL;
+	}
+
```

Comment:
this code is only called when additional info is needed to decide query's distribution policy (like, when it's inherited from patrent table). it doesn't sits in the normal code path, because normally the `DISTRIBUTED BY` would already be parsed by the parser and stored in `CreateStmt->distributedBy`. I guess this is for https://github.com/cloudberrydb/cloudberrydb/pull/206#discussion_r1335544239 because utility supports inheritance now and this may be called in the case.

Reply:
places where `make_distributedby_for_rel` are called:
- `alter table set distributed by`: not supported in utility
- `alter table expand`: ditto
- `create table distributed by`: ditto
- `create table partition by`: when creating partitioned table in utility mode (yes, singlenode already supports partitioned table), because `distributed by` is meaningless, there's no need to deduce distribution policy from the clause, we can always return a NULL. and because creating table **under utility mode** is the only case when this function may be called, we can safely assert that `policy->ptype` is `ENTRY`. as described above, there won't be a case where table get created in cluster mode but altered in singlenode or utility mode.